src/main/java/org/apache/pulsar/io/jcloud/sink/BlobStoreAbstractSink.java

```diff
@@ -126,7 +126,7 @@ private void flushIfNeeded(boolean force) {
             return;
         }
         if (force || currentBatchSize.get() >= maxBatchSize || currentBatchBytes.get() >= maxBatchBytes) {
-            flushExecutor.submit(this::flush);
+            flushExecutor.submit(() -> flush(false));
```

Comment:
We need to pass `force` into the `flush(bool fource)` method, right?

When connector call close(), will use force=true to call `flushIfNeeded `.

If we change it to this, it will skip flush().

```
        if (!force && currentBatchSize.get() < maxBatchSize && currentBatchBytes.get() < maxBatchBytes) {
            log.debug("Skip flushing because the batch is not full.");
            return;
        }
```